Abbyy.CloudSdk.V2.Client.Sample/Program.cs

```diff
@@ -66,6 +72,47 @@ private static async Task<List<string>> ProcessImageAsync()
 
 				return taskInfo.ResultUrls;
 			}
-	    }
+		}
+
+		private static async Task<TaskList> GetFinishedTasksWithRetry()
+		{
+			var retryCount = 3;
+			var millisecondsDelay = 3000;
+
+			IAsyncPolicy<HttpResponseMessage>[] retryPolicies =
+			{
+				Policy
+					.HandleResult<HttpResponseMessage>(result => result.StatusCode == HttpStatusCode.GatewayTimeout)
+					.WaitAndRetryAsync(
+						retryCount,
+						sleepDurationProvider => TimeSpan.FromMilliseconds(millisecondsDelay),
+						onRetry: (exception, calculatedWaitDuration, retries, context) =>
+						{
+							Console.WriteLine($"Retry {retries} for policy with key {context.PolicyKey}");
+						}
+					)
+					.WithPolicyKey("WaitAndRetryAsync_For_GatewayTimeout_504__StatusCode")
+			};
+
+			// Here it is necessary to take into account the maximum delay time of the policies,
+			// so that the request does not expire on timeout earlier than it should
+			var delayForRetry = millisecondsDelay * retryCount;
+
+			using (var httpClientHandlerWithRetry = new HttpClientHandlerWithRetry(_authInfo, retryPolicies))
+			using (var httpClient = new HttpClientWithRetry(_authInfo, httpClientHandlerWithRetry, delayForRetry))
+			using (var client = new OcrClientWithRetry(httpClient))
+			{
+				try
+				{
+					var finishedTasks = await client.ListFinishedTasksAsync();
+					return finishedTasks;
+				}
+				catch (Exception ex)
+				{
+					Console.WriteLine(ex);
```

Comment:
Maybe we should remove `try/catch` and `Console.Writeline` / `return new TaskList();` from this example.
It's helpful for testing purpose, but don't needed as example.